tests/unit/steps/test_hypervisor.py

```diff
@@ -268,7 +286,95 @@ def test_hypervisor_upgrade_plan(model):
         workload_version="21.0.0",
     )
 
-    planner = HypervisorUpgradePlanner([cinder, nova_compute])
+    planner = HypervisorUpgradePlanner([cinder, nova_compute], list(machines.values()))
+    plan = planner.generate_upgrade_plan(target, False)
+
+    assert str(plan) == exp_plan
+
+
+def test_hypervisor_upgrade_plan_filtered(model):
```

Comment:
not-blocking: `test_hypervisor_upgrade_plan_filtered` -> `test_hypervisor_upgrade_plan_single_machine` or maybe even drop `test_hypervisor_plan_single_unit` (Meaning we need to do same for test above).